ui/nuxeo-page-provider-display-behavior.js

```diff
@@ -392,7 +412,12 @@ export const PageProviderDisplayBehavior = [
 
     _isSelected(item) {
       return this.multiSelection
-        ? !!(this.selectedItems && this.selectedItems.length && this.selectedItems.indexOf(item) > -1)
+        ? !!(
+            this.selectedItems &&
+            this.selectedItems.length &&
+            this.selectedItems.indexOf(item) > -1 &&
+            !this._excludedItems.includes(item.uid)
+          ) || !!(this._isSelectAllActive && this._excludedItems.length && !this._excludedItems.includes(item.uid))
```

Comment:
The _isSelected check seems to be causing the issue we are seeing with the selection, In some cases, it seems to be that the item instance might have changed, which cause indexOf to tail. Nonetheless, simplifying this a bit also seems to solve the issue. Please give this a try (it also contains some cleanup):
```diff
diff --git a/ui/nuxeo-page-provider-display-behavior.js b/ui/nuxeo-page-provider-display-behavior.js
index 0ef88461..55d2c130 100644
--- a/ui/nuxeo-page-provider-display-behavior.js
+++ b/ui/nuxeo-page-provider-display-behavior.js
@@ -311,12 +311,10 @@ export const PageProviderDisplayBehavior = [
         if (this.selectAllActive && this.items[index]) {
           if (!this._excludedItems.includes(this.items[index].uid)) {
             this.push('_excludedItems', this.items[index].uid);
-            this._selectItemModel(index);
             return;
           }
           if (this._excludedItems.includes(this.items[index].uid)) {
             this._excludedItems = this._excludedItems.filter((item) => item !== this.items[index].uid);
-            this._selectItemModel(index);
           }
         }
       }
@@ -411,13 +409,11 @@ export const PageProviderDisplayBehavior = [
     },
 
     _isSelected(item) {
+      if (this._isSelectAllActive) {
+        return !this._excludedItems.includes(item.uid);
+      }
       return this.multiSelection
-        ? !!(
-            this.selectedItems &&
-            this.selectedItems.length &&
-            this.selectedItems.indexOf(item) > -1 &&
-            !this._excludedItems.includes(item.uid)
-          ) || !!(this._isSelectAllActive && this._excludedItems.length && !this._excludedItems.includes(item.uid))
+        ? !!(this.selectedItems && this.selectedItems.length && this.selectedItems.indexOf(item) > -1)
         : !!(this.selectedItem && this.selectedItem === item);
     },
 
@@ -831,20 +827,21 @@ export const PageProviderDisplayBehavior = [
     _selectItemModel(index) {
       if (this.$.list._isIndexRendered(index)) {
         const model = this.modelForElement(this.$.list._physicalItems[this.$.list._getPhysicalIndex(index)]);
+        const item = this.items[index];
+        if (!item) {
+          return;
+        }
         if (model) {
           if (
             !this._excludedItems ||
             !this._excludedItems.length ||
-            (this._excludedItems.length && this.items[index] && !this._excludedItems.includes(this.items[index].uid))
+            (this._excludedItems.length && !this._excludedItems.includes(item.uid))
           ) {
             model[this.$.list.selectedAs] = true;
-            return;
-          }
-          if (
+          } else if (
             this._excludedItems &&
             this._excludedItems.length &&
-            this.items[index] &&
-            this._excludedItems.includes(this.items[index].uid)
+            this._excludedItems.includes(item.uid)
           ) {
             model[this.$.list.selectedAs] = false;
           }
```